### ISSUE TITLE
call back in emit not working

### ISSUE THREAD

**Issue by Renjila** [miguelgrinberg/python-socketio#525]
I have given an emit  function in client as follows

     def cb_func(data):
		result= json.loads(data)
		print (result)
		flag = True
		sio.disconnect()
        @sio.event
	def connect():
		print('connected')
		print('client: xxxxx')
		sio.emit('client:startCheck',data={'testId': 12, 'externalId': 1},callback=cb_func) 

Here the call back function is not getting called. 
Please let me know how the call back function gets called in emit.
Thanks

**Comment by miguelgrinberg** [comment]
Can you provide a complete example that I can test here, or at least logs for a run of your application?

**Comment by RenjilaRajan** [comment]
Its working now. 
When I first checked, the disconnect method was getting called soon after the connect.
Then I tried another socket io client for python (socketIO-client 0.7.2 from pypi.org), but couldn't get it working and left it after trying only a bit.
After that I created a local socket io server using the first code from Server Examples here https://python-socketio.readthedocs.io/en/latest/intro.html and tried connect and emit to this server .For this I installed eventlet (pip3 install eventlet). It was working fine.
And then when I went back and tried my original code connecting to actual remote server, it was working suddenly, without any change in the code. Not sure what made it work, and what was the cause for it to not work earlier. Could the installation of two python libraries I mentioned above (or it is dependencies) have any effect on this?
